memtable.cc

```diff
@@ -741,23 +741,26 @@ memtable::make_flat_reader(schema_ptr s,
         rd.upgrade_schema(s);
         return rd;
     } else {
-        auto res = make_flat_mutation_reader<scanning_reader>(std::move(s), shared_from_this(), std::move(permit), range, slice, pc, fwd_mr);
+        flat_mutation_reader_v2 res = make_flat_mutation_reader_v2<scanning_reader>(std::move(s), shared_from_this(), std::move(permit), range, slice, pc, fwd_mr);
+        auto res_v1 = downgrade_to_v1(std::move(res));
         if (fwd == streamed_mutation::forwarding::yes) {
-            return make_forwardable(std::move(res));
+            return make_forwardable(std::move(res_v1));
         } else {
-            return res;
+            return res_v1;
         }
     }
 }
 
 flat_mutation_reader
 memtable::make_flush_reader(schema_ptr s, reader_permit permit, const io_priority_class& pc) {
     if (group()) {
-        return make_flat_mutation_reader<flush_reader>(std::move(s), std::move(permit), shared_from_this());
+        auto reader = make_flat_mutation_reader_v2<flush_reader>(std::move(s), std::move(permit), shared_from_this());
+        return downgrade_to_v1(std::move(reader));
     } else {
         auto& full_slice = s->full_slice();
-        return make_flat_mutation_reader<scanning_reader>(std::move(s), shared_from_this(), std::move(permit),
-            query::full_partition_range, full_slice, pc, mutation_reader::forwarding::no);
+        auto reader = make_flat_mutation_reader_v2<scanning_reader>(std::move(s), shared_from_this(), std::move(permit),
+                      query::full_partition_range, full_slice, pc, mutation_reader::forwarding::no);
+        return downgrade_to_v1(std::move(reader));
```

Comment:
This is sad, is there a performance impact?

Reply:
I highly doubt it improves performance, indeed.
I didn't do any measurements, perhaps @enedil did when he authored this?

Reply:
No, I didn't test it then.

Reply:
Once "[PATCH v1 00/17] Migrate sstable writer to v2" goes in, we can switch to a `make_flush_reader_v2()` and get rid of this downgrade.